pkg/autoupdate/tuf/channel_version.go

```diff
@@ -0,0 +1,33 @@
+package tuf
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/kolide/launcher/pkg/agent"
+)
+
+// VersionForChannel returns the tagged version of the given binary for the given release channel.
+func VersionForChannel(binary, channel, tufServerUrl string) (string, error) {
```

Comment:
What calls this:

> VersionForChannel to be used by launcher doctor and by packaging changes for https://github.com/kolide/launcher/issues/1149

Discussion on the launcher doctor PR: https://github.com/kolide/launcher/pull/1197#discussion_r1195584997

Since package build doesn't necessarily run on a machine with launcher installed, it can't use the populated TUF repo -- and either way, this feels a little safer than having separate routines updating the same TUF directory potentially at the same time.

I have a TODO to ship a more complete/up-to-date TUF repo -- hopefully that addresses the performance concern some?

Reply:
_ah_

So it's not used by this PR, and will be used by things that operate occasionally (package build, doctor). Given their nature, they need to talk to the server, and the additional overhead is okay. That makes sense.

I think we should name this something more clear, and godoc it appropriately. Maybe `GetVersionFromTufServer` or something?

Reply:
Yup I like that better, I will rename + add some docs!